hermes/probe/monitor.go

```diff
@@ -12,9 +12,160 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
+// Authors:
+// - Alicja Kwiecinska (kwiecinskaa@google.com) github: alicjakwie
+// - Evan Spendlove, GitHub: evanSpendlove.
+//
 // Monitor implements a probe that monitors a storage system using the
 // Hermes algorithm.
+// Package probe implements the probe that Hermes uses to monitor
+// a storage system.
+
+package probe
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/google/cloudprober/logger"
+	cpmetrics "github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+
+	probepb "github.com/googleinterns/step224-2020/config/proto"
+	journalpb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+// Target holds all of the required information and state for a given target run.
+type Target struct {
+	// Target stores the proto config for the target to be probed.
+	Target *probepb.Target
+
+	// Journal stores the state of MonitorProbe as a combination of a next operation intent enum and a filenames map.
+	Journal *journalpb.StateJournal
+
+	// LatencyMetrics stores the api call and probe operation latency for a given target run.
+	// Metrics are stored with additional labels to record operation type and exit status.
+	LatencyMetrics *metrics.Metrics
+}
 
 // MonitorProbe holds aggregate information about all probe runs, per-target.
 // It also holds the config and options used to initialise the probe.
-type MonitorProbe struct{}
+type MonitorProbe struct {
+	name    string
+	config  *probepb.HermesProbeDef
+	targets []*Target
+	opts    *options.Options
+	logger  *logger.Logger
+}
+
+// Init initializes the probe with the given parameters.
+// This is a required method to implement the cloudprober.Probes.Probe interface.
+func (mp *MonitorProbe) Init(name string, opts *options.Options) error {
+	// Cast from Cloudprober defined interface to HermesProbeDef.
+	// This allows for accessing the variables and methods of a HermesProbeDef object.
+	conf, ok := opts.ProbeConf.(*probepb.HermesProbeDef)
+	if !ok {
+		return fmt.Errorf("invalid argument: opts.ProbeConf is not of type *probepb.HermesProbeDef")
+	}
+	mp.name = name
+	mp.config = conf
+
+	mp.targets = make([]*Target, 0)
+	for _, t := range mp.config.GetTargets() {
+		run := &Target{
```

Comment:
No need for the intermediate variable.
Just
```
m.targets = append(m.targets, &Target{
...
})

Reply:
I've amended this.